internal/provision/profiles.go

```diff
@@ -65,38 +56,28 @@ func LoadProfiles(path string, dic *di.Container) errors.EdgeX {
 			fullPath := filepath.Join(absPath, fName)
 			yamlFile, err := ioutil.ReadFile(fullPath)
 			if err != nil {
-				lc.Errorf("failed to read %s: %v", fullPath, err)
+				lc.Errorf("Failed to read %s: %v", fullPath, err)
 				continue
 			}
 
 			err = yaml.Unmarshal(yamlFile, &profile)
 			if err != nil {
-				lc.Errorf("filed to decode profile %s: %v", fullPath, err)
-				continue
-			}
-			// if profile already exists in metadata, skip it
-			if p, ok := profileMap[profile.Name]; ok {
-				_ = cache.Profiles().Add(p)
+				lc.Errorf("Failed to decode profile %s: %v", fullPath, err)
 				continue
 			}
 
-			// add profile to metadata
-			ctx := context.WithValue(context.Background(), common.CorrelationHeader, uuid.NewString())
-			_, err = dpc.AddByYaml(ctx, fullPath)
-			if err != nil {
-				lc.Errorf("failed to add profile %s to metadata: %v", fullPath, err)
-				continue
+			_, err = dpc.DeviceProfileByName(context.Background(), profile.Name)
+			if err == nil {
+				lc.Infof("Profile %s exists, using the existing one", profile.Name)
+			} else {
+				lc.Infof("Profile %s not found in Metadata, adding it ...", profile.Name)
+				ctx := context.WithValue(context.Background(), common.CorrelationHeader, uuid.NewString())
+				_, err = dpc.AddByYaml(ctx, fullPath)
```

Comment:
Since the YAML has been parsed, call add by profile dto func because core-metadata needs to parse YAML and transfer to JSON again.

Reply:
Also, the profiles can be created in one request